app/src/main/java/org/mifos/mobile/ui/loan_account_withdraw/LoanAccountWithdrawScreen.kt

```diff
@@ -0,0 +1,198 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.tooling.preview.UiMode
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosProgressIndicator
+import org.mifos.mobile.core.ui.component.MifosTitleDescSingleLineEqual
+import org.mifos.mobile.core.ui.component.MifosTopBar
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.models.accounts.loan.LoanWithAssociations
+import org.mifos.mobile.ui.savings_account_withdraw.ErrorComponent
+
+
+
+@Composable
+fun LoanAccountWithdrawScreen(
+    viewModel: LoanAccountWithdrawViewModel = hiltViewModel(),
+    navigateBack: () -> Unit,
+    withdraw: (String) -> Unit,
+) {
+    val uiState by viewModel.loanUiState.collectAsStateWithLifecycle()
+    val loanWithAssociations by viewModel.loanWithAssociations.collectAsStateWithLifecycle()
+
+    LoanAccountWithdrawScreen(
+        uiState = uiState,
+        loanWithAssociations = loanWithAssociations,
+        navigateBack = navigateBack,
+        withdraw = withdraw,
+    )
+}
+
+
+@Composable
+fun LoanAccountWithdrawScreen(
+    uiState: LoanAccountWithdrawUiState,
+    loanWithAssociations: LoanWithAssociations?,
+    navigateBack: () -> Unit,
+    withdraw: (String) -> Unit
+) {
+    val context = LocalContext.current
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.background)
+    ) {
+        MifosTopBar(
+            navigateBack = { navigateBack() },
+            title = { Text(text = stringResource(id = R.string.withdraw_loan)) }
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Box(modifier= Modifier.weight(1f)) {
+            LoanAccountWithdrawContent(
+                loanWithAssociations = loanWithAssociations,
+                withdraw = withdraw
+            )
+
+            when (uiState) {
+                is LoanAccountWithdrawUiState.Loading -> {
+                    MifosProgressIndicator(modifier = Modifier
+                        .fillMaxSize()
+                        .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                    )
+                }
+
+                is LoanAccountWithdrawUiState.Success -> {
+                    Toast.makeText(context, R.string.loan_application_withdrawn_successfully, Toast.LENGTH_SHORT).show()
+                    navigateBack()
+                }
+
+                is LoanAccountWithdrawUiState.Error -> {
+                    ErrorComponent(errorToast = stringResource(id = uiState.messageId))
+                }
+
+                is LoanAccountWithdrawUiState.WithdrawUiReady -> Unit
```

Comment:
We can remove this state seems useless/.

Reply:
I am using this state as a default one in the MutableStateFlow. What to keep as the default state if I removed this? Loading, Success or Error will be called only after withdraw button pressed.